controllers/argocdmetrics/argocd_metrics_controller.go

```diff
@@ -34,49 +48,29 @@ var logs = logf.Log.WithName("controller_argocd_metrics")
 type ArgoCDMetricsReconciler struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	client client.Client
-	scheme *runtime.Scheme
+	Client client.Client
+	Scheme *runtime.Scheme
 }
 
 // blank assignment to verify that ReconcileArgoCDRoute implements reconcile.Reconciler
 var _ reconcile.Reconciler = &ArgoCDMetricsReconciler{}
 
-// Add creates a new ArgoCD Metrics Controller and adds it to the Manager.
-// The Manager will set fields on the Controller and start it when the
-// Manager is Started.
-func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+// SetupWithManager sets up the controller with the Manager.
+func (r *ArgoCDMetricsReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&argoapp.ArgoCD{}).
+		Complete(r)
 }
 
-func add(mgr manager.Manager, reconciler reconcile.Reconciler) error {
-	reqLogger := logs.WithValues()
-
-	reqLogger.Info("Creating ArgoCD metrics controller")
-	c, err := controller.New("argocd-metrics-controller", mgr, controller.Options{Reconciler: reconciler})
-	if err != nil {
-		return err
-	}
-
-	reqLogger.Info("Watching for ArgoCD instance creation")
-	err = c.Watch(&source.Kind{Type: &argoapp.ArgoCD{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ArgoCDMetricsReconciler{client: mgr.GetClient(), scheme: mgr.GetScheme()}
-}
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=prometheuses;prometheusrules;servicemonitors,verbs=*
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=servicemonitors,verbs=create;get
 
 func (r *ArgoCDMetricsReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := logs.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling ArgoCD Metrics")
 
 	namespace := corev1.Namespace{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: request.Namespace}, &namespace)
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: request.Namespace}, &namespace)
```

Comment:
When the new `Reconcile(context.Context, reconciler.Request) (reconciler.Result, error)` method is implemented, the calls to `context.TODO()` should be changed to use the `ctx` parameter passed in.